be/test/common/s3_uri_test.cpp

```diff
@@ -41,13 +41,42 @@ PARALLEL_TEST(S3URITest, path_style_url) {
 }
 
 PARALLEL_TEST(S3URITest, s3_scheme) {
-    S3URI uri;
-    ASSERT_TRUE(uri.parse("s3://mybucket/puppy.jpg"));
-    EXPECT_EQ("s3", uri.scheme());
-    EXPECT_EQ("", uri.region());
-    EXPECT_EQ("mybucket", uri.bucket());
-    EXPECT_EQ("puppy.jpg", uri.key());
-    EXPECT_EQ("", uri.endpoint());
+    {
+        S3URI uri;
+        ASSERT_TRUE(uri.parse("s3://mybucket/puppy.jpg"));
+        EXPECT_EQ("s3", uri.scheme());
+        EXPECT_EQ("", uri.region());
+        EXPECT_EQ("mybucket", uri.bucket());
+        EXPECT_EQ("puppy.jpg", uri.key());
+        EXPECT_EQ("", uri.endpoint());
+    }
+    {
+        S3URI uri;
+        ASSERT_TRUE(uri.parse("S3://test.name.with.dot/another.test/foo.parquet"));
```

Comment:
how about `https://my.s3.bucket.s3.us-east-1.amazonaws.com`? not sure if `my.s3.bucket` is allowed as a bucket name.